### PR TITLE
compiler: add option `@generated=omit`

### PR DESCRIPTION
**Pull request by panchenko**
related to #11081

### CODE REVIEW

**Comment by alexanderankin** [comment]
@panchenko is this backwards compatible, as people are already using this?

**Comment by panchenko** [comment]
> @panchenko is this backwards compatible, as people are already using this?

The option with an unclear name has been released just a few days ago, it can be discussed if it's worth keeping it compatible, personally I don't think so.
IMHO can be handled like a "typo".

**Comment by alexanderankin** [comment]
@panchenko - even so, you have examples of people already using it in the thread, and many others who got notified on several email threads (not just this repo) about the issue being closed (and reopened).

besides that on merit, the name jakarta makes more sense as this is what new programmers are coming in learning about. the name javax doesn't mean anything anymore, jakarta_include (which is unlikely to be implemented but one can hope), jakarta_omit, and jakarta_javax make sense, but including = does not because these are not key value pairs, they are really just `int argc, char *argv[]`, which is misleading to users of the library, which, since its java, are just people at work whose lives are already made miserable without our help

**Review by ejona86 — APPROVED** [review]
I like this. I also wasn't wild about the option name and delayed string checking before.

I have some concern of using `@` at the beginning of an option, as `@filename` is special in protoc and elsewhere. But it looks like it'd be fine.

**Comment by ejona86** [comment]
> panchenko is this backwards compatible, as people are already using this?

I'm not too worried about backward compatibility because the users setting this are also in control of the protoc-gen-grpc version. It's also only been around a single release.

> name jakarta makes more sense as this is what new programmers are coming in learning about

The chance we ever use the Jakarta classes here is close to 0%.

> the name javax doesn't mean anything anymore

It is literally in the package name for the class. And that's also only because you are aware of what is and is not part of Jakarta. javax itself is still very much alive: javax.crypto, javax.net, javax.xml....

> including = does not because these are not key value pairs, they are really just int argc, char *argv[]

That is identical to saying a command line argument `--foo=bar` doesn't make sense, even though it is frequently used. The options are weird in protobuf as they were bolted-on, but using equals is standard (just like how --baz or --foo=bar are both normal). https://github.com/protocolbuffers/protobuf/blob/da651cd3c286d0f2206f13947d96422370fa5ccc/src/google/protobuf/compiler/code_generator.h#L224

File: compiler/src/testLite/golden/TestDeprecatedService.java.txt [hunk 1]
```diff
@@ -7,9 +7,6 @@ import static io.grpc.MethodDescriptor.generateFullMethodName;
  * Test service that has been deprecated and should generate with Java's &#64;Deprecated annotation
  * </pre>
  */
-@javax.annotation.Generated(
```
Comment (sergiitk) [inline compiler/src/testLite/golden/TestDeprecatedService.java.txt:10]: @ejona86 - do you think we need another variant of this test to cover the case where `@javax.annotation.Generated` is not skipped?

Comment (panchenko) [reply compiler/src/testLite/golden/TestDeprecatedService.java.txt:10]: That's kept in `test/golden`
https://github.com/grpc/grpc-java/blob/add8c37a4148f6dd857dd31c7425e2ae526e0528/compiler/src/test/golden/TestDeprecatedService.java.txt#L10-L15

Comment (ejona86) [reply compiler/src/testLite/golden/TestDeprecatedService.java.txt:10]: Yeah, we're "abusing" the fact we have lite vs normal and only doing this is the lite version.

Comment (sergiitk) [reply compiler/src/testLite/golden/TestDeprecatedService.java.txt:10]: Ah, I see.

**Review by sergiitk — APPROVED** [review]

**Pull request merged**